src/app/shared/preview-results/preview-results.component.ts

```diff
@@ -0,0 +1,142 @@
+import {QueryModel} from "../../core/store/navigation/query.model";
+import {Workspace} from "../../core/store/navigation/workspace.model";
+import {DocumentsAction} from "../../core/store/documents/documents.action";
+
+@Component({
+  selector: 'preview-results',
+  templateUrl: './preview-results.component.html',
+  styleUrls: ['./preview-results.component.scss']
+})
+export class PreviewResultsComponent implements OnInit, OnDestroy {
+
+  public selectedCollectionId: string;
+
+  public collections$: Observable<CollectionModel[]>;
+
+  public documents$: Observable<DocumentModel[]>;
+
+  public collection$: Observable<CollectionModel>;
+
+  private allSubscriptions = new Subscription();
+
+  private query: QueryModel;
+
+  private collectionQuery: QueryModel;
+
+  @Output()
+  public selectCollection = new EventEmitter<CollectionModel>();
+
+  @Output()
+  public selectDocument = new EventEmitter<DocumentModel>();
+
+  private collectionsCount = -1;
+  private documentsCount = -1;
+
+  constructor(private store: Store<AppState>) { }
+
+  public ngOnInit() {
+    this.subscribeAll();
+  }
+
+  public ngOnDestroy() {
+    this.unsubscribeAll();
+  }
+
+  private subscribeAll() {
+    this.collections$ = this.store.select(selectCollectionsByQuery).pipe(
+      filter(collections => !isNullOrUndefined(collections) && collections.length > 0)
+    );
+
+    this.allSubscriptions.add(this.store.select(selectNavigation).pipe(
+      filter(navigation => this.validWorkspace(navigation.workspace))
+    ).subscribe(navigation => this.updateNavigation(navigation.query)));
+
+    // initialize when we do not select anything
+    this.allSubscriptions.add(this.collections$.pipe(filter(collections => !!collections), take(1))
+      .subscribe(collections => {
+        this.collectionsCount = collections.length;
+        if (!this.selectedCollectionId) {
+          this.setActiveCollection(collections[0]);
+        }
+      }));
+  }
+
+  private updateNavigation(query: QueryModel) {
+    this.query = query;
+    this.getData();
+  }
+
+  private validWorkspace(workspace: Workspace): boolean {
+    if (workspace && workspace.organizationCode && workspace.projectCode) {
+      return true;
+    }
+    return false;
+  }
+
+  private unsubscribeAll() {
+    this.allSubscriptions.unsubscribe();
+  }
+
+  public setActiveCollection(collection: CollectionModel) {
+    this.selectedCollectionId = collection.id;
+    this.getData();
+    this.selectCollection.emit(collection);
+  }
+
+  private updateDataSubscription() {
+    if (this.collectionQuery) {
+      this.documents$ = this.store.select(selectDocumentsByCustomQuery(this.collectionQuery));
+      this.collection$ = this.store.select(selectCollectionById(this.selectedCollectionId));
+
+      this.allSubscriptions.add(this.documents$.pipe(filter(documents => !!documents && documents.length > 0),
+        tap(documents => {
+          this.documentsCount = documents.length;
+        }),
+        take(1))
```

Comment:
Is it an intention or a mistake that the `documentsCount` will be updated every time this pipe is executed but only for the first item the `subscribe` method will be executed?

Reply:
This is correct. Purpose of this subscription is twofold :-)